src/Microsoft.Diagnostics.Monitoring.RestServer/ProcessFilterOptions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+#if UNITTEST
+namespace Microsoft.Diagnostics.Monitoring.UnitTests.Options
+#else
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+#endif
+{
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public enum ProcessFilterKey
+    {
+        Pid,
+        RuntimeId,
+        Name,
+        CommandLine,
+    }
+
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public enum ProcessFilterType
+    {
+        [Display(Description = "Performs a case-insensitive string comparison.")]
+        Exact,
+
+        [Display(Description = "Performs a case-insensitive substring search.")]
+        Loose,
+    }
+
+    public sealed class ProcessFilterOptions
+    {
+        [Display(Description = "Process filters used to determine the process when collecting metrics. All filters must match.")]
+        public List<ProcessFilterDescriptor> ProcessFilters { get; set; } = new List<ProcessFilterDescriptor>(0);
```

Comment:
nit: just call it `Filters`. Given that this property is on a class called `ProcessFilterOptions` and the options in the configuration are under `DefaultProcess`, I don't think repeating Process here adds anything.